firebase-perf/src/main/java/com/google/firebase/perf/metrics/AppStartTrace.java

```diff
@@ -404,26 +404,31 @@ private void logAppStartTrace() {
             .setDurationUs(getClassLoadTimeCompat().getDurationMicros(onResumeTime));
     List<TraceMetric> subtraces = new ArrayList<>(/* initialCapacity= */ 3);
 
-    TraceMetric.Builder traceMetricBuilder =
-        TraceMetric.newBuilder()
-            .setName(Constants.TraceNames.ON_CREATE_TRACE_NAME.toString())
-            .setClientStartTimeUs(getClassLoadTimeCompat().getMicros())
-            .setDurationUs(getClassLoadTimeCompat().getDurationMicros(onCreateTime));
-    subtraces.add(traceMetricBuilder.build());
-
-    traceMetricBuilder = TraceMetric.newBuilder();
-    traceMetricBuilder
-        .setName(Constants.TraceNames.ON_START_TRACE_NAME.toString())
-        .setClientStartTimeUs(onCreateTime.getMicros())
-        .setDurationUs(onCreateTime.getDurationMicros(onStartTime));
-    subtraces.add(traceMetricBuilder.build());
-
-    traceMetricBuilder = TraceMetric.newBuilder();
-    traceMetricBuilder
-        .setName(Constants.TraceNames.ON_RESUME_TRACE_NAME.toString())
-        .setClientStartTimeUs(onStartTime.getMicros())
-        .setDurationUs(onStartTime.getDurationMicros(onResumeTime));
-    subtraces.add(traceMetricBuilder.build());
+    TraceMetric.Builder traceMetricBuilder = TraceMetric.newBuilder()
+              .setName(Constants.TraceNames.ON_CREATE_TRACE_NAME.toString())
+              .setClientStartTimeUs(getClassLoadTimeCompat().getMicros())
+              .setDurationUs(getClassLoadTimeCompat().getDurationMicros(onCreateTime));
```

Comment:
Is it possible for `onCreateTime` to be null? Eg. Could `StartFromBackgroundRunnable` be triggered before onCreate even gets called? The comment around it suggests so. If so, we will need more if to cover `ON_CREATE_TRACE_NAME` and `ON_START_TRACE_NAME`

Reply:
In case of app starting from background using `StartFromBackgroundRunnable`, logAppStartTrace would not even be called since we do not dispatch app start trace in those situations. So, this code path would not even executed in that case. Essentially, `OnActivityResumed` will not be even called. If `OnActivityResumed` is called, it is fair to assume that `OnActivityCreate` would be called.

Reply:
Marking it as resolved. Please feel free to reopen if you disagree with my response.